framework/helpers/BaseJson.php

```diff
@@ -126,28 +127,25 @@ public static function decode($json, $asArray = true)
     /**
      * Handles [[encode()]] and [[decode()]] errors by throwing exceptions with the respective error message.
      *
-     * @param int $lastError error code from [json_last_error()](https://secure.php.net/manual/en/function.json-last-error.php).
+     * @param int $lastError error code from [json_last_error()](https://www.php.net/manual/en/function.json-last-error.php).
      * @throws InvalidArgumentException if there is any encoding/decoding error.
      * @since 2.0.6
      */
     protected static function handleJsonError($lastError)
     {
-        if ($lastError === JSON_ERROR_NONE) {
-            return;
-        }
+        if ($lastError !== JSON_ERROR_NONE) {
+            if (PHP_VERSION_ID >= 50500) {
```

Comment:
Does that make all constants starting from PHP 5.5 useless?

Reply:
constants contains error codes, not messages

Reply:
Yes but we aren't using any error codes if PHP version is more recent than 5.5.

Reply:
@samdark https://www.php.net/manual/en/function.json-last-error-msg.php#117393

Reply:
So? The code reads:

```
int lastError
if (lastError is not NONE and PHP >= 5.5) {
  throw exception using last error message (not the code)
}

the rest of the logic for lastError = NONE or PHP < 5.5
```

Reply:
@samdark 
~~~
if (lastErrorCode is not JSON_ERROR_NONE) {
  throw exception(PHP >= 5.5 ? json-last-error-msg() : jsonErrorMessages[lastErrorCode], lastErrorCode);
}
~~~
no logic for lastErrorCode = JSON_ERROR_NONE, because... no errors...